### DISCUSSION TITLE
populate neo4j locally

### DISCUSSION THREAD

**Discussion by johaGL** [MetabolicAtlas/MetabolicAtlas#1289] [Q&A] [answered]
Dear MetabolicAtlas team, thank you for your great software. I have installed Neo4j locally and I would like to populate it following your instructions. You explain in the readme.md that the MetabolicAtlas,  data-generation and data-files repos are required, but please correct me if I am wrong: your instructions are intended to deploy the entire Metabolic Atlas web with front-end + back-end + Neo4j.  My purpose is just to have locally a populated Neo4j  for me to exploit it programmatically. Could you please tell me if  data-generation and data-files are enough  for my purpose ? Thank you in advance for your time and help.

**Comment by mihai-sysbio** [comment, marked as answer]
Hi @johaGL!

When installing Neo4j locally, I presume this is without Docker.
Based on your use-case, I believe it would be enough to have the `data-generation` and `data-files` repositories. This [Readme](https://github.com/MetabolicAtlas/data-generation#generating-data) is the optimal stating point. However, this will only _generate_ the data in the correct format to be imported by Neo4j. The command to import the generated data into Neo4j is hidden [here](https://github.com/MetabolicAtlas/MetabolicAtlas/blob/f37d488a3899127beaf8537beb278095f1b28b75/proj.sh#L70):

```
cypher-shell -u ${NEO4J_USERNAME} -p ${NEO4J_PASSWORD} --format plain --file import/import.cypher
```

where `import.cypher` is generated to contain the instructions to import all other generated CSV files. Feel free to define your own username and password; for us, those have to be variables that get used elsewhere too.

I hope this helps.

**Reply by johaGL** [reply]
Thank you very much!
